src/array.jl

```diff
@@ -1050,3 +1050,27 @@ StructTypes.construct(::Type{<:CategoricalArray{Union{Nothing, T}}},
                       A::Vector) where {T} =
     categoricalnothing(T, A)
 categoricalnothing(T, A::AbstractVector) = CategoricalArray{Union{Nothing, T}}(A)
+
+# DataAPI refarray/refvalue/refpool support
+struct CategoricalRefPool{T, P} <: AbstractVector{T}
+    pool::P
+end
+
+Base.IndexStyle(::Type{<: CategoricalRefPool}) = Base.IndexLinear()
+@inline function Base.getindex(x::CategoricalRefPool, i::Int)
```

Comment:
A minor thing - don't we need `@propagate_inbounds` here?

Reply:
Not AFAIK. It's only needed when `@inbounds` should be propagated to a function that is called from there. But here we use `@inbounds` explicitly.